packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js

```diff
@@ -0,0 +1,189 @@
+ */
+import { Platform, findNodeHandle } from 'react-native';
+import { partial, first, castArray, last, compact } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { ToolbarButton, Picker } from '@wordpress/components';
+import { getBlockType, getDefaultBlockName } from '@wordpress/blocks';
+import { __, sprintf } from '@wordpress/i18n';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { withInstanceId, compose } from '@wordpress/compose';
+import { moreHorizontalMobile, trash, cog } from '@wordpress/icons';
+/**
+ * Internal dependencies
+ */
+import { getMoversSetup } from '../block-mover/mover-description';
+
+const BlockActionsMenu = ( {
+	onDelete,
+	isStackedHorizontally,
+	wrapBlockSettings,
+	wrapBlockMover,
+	openGeneralSidebar,
+	onMoveDown,
+	onMoveUp,
+	isFirst,
+	isLast,
+	blockTitle,
+	isEmptyDefaultBlock,
+	blockMobileToolbarRef,
+} ) => {
+	const moversOptions = { keys: [ 'icon', 'actionTitle' ], blockTitle };
+
+	const {
+		icon: { backward: backwardButtonIcon, forward: forwardButtonIcon },
+		actionTitle: {
+			backward: backwardButtonTitle,
+			forward: forwardButtonTitle,
+		},
+	} = getMoversSetup( isStackedHorizontally, moversOptions );
+
+	const deleteOption = {
+		id: 'deleteOption',
+		// translators: %s: block title e.g: "Paragraph".
+		label: sprintf( __( 'Remove %s' ), blockTitle ),
+		value: 'deleteOption',
+		icon: trash,
+		separated: true,
+		disabled: isEmptyDefaultBlock,
+	};
+
+	const settingsOption = {
+		id: 'settingsOption',
+		// translators: %s: block title e.g: "Paragraph".
+		label: sprintf( __( '%s Settings' ), blockTitle ),
+		value: 'settingsOption',
+		icon: cog,
+	};
+
+	const backwardButtonOption = {
+		id: 'backwardButtonOption',
+		label: backwardButtonTitle,
+		value: 'backwardButtonOption',
+		icon: backwardButtonIcon,
+		disabled: isFirst,
+	};
+
+	const forwardButtonOption = {
+		id: 'forwardButtonOption',
+		label: forwardButtonTitle,
+		value: 'forwardButtonOption',
+		icon: forwardButtonIcon,
+		disabled: isLast,
+	};
+
+	const options = compact( [
+		wrapBlockMover && backwardButtonOption,
+		wrapBlockMover && forwardButtonOption,
+		wrapBlockSettings && settingsOption,
+		deleteOption,
+	] );
+
+	function onPickerSelect( value ) {
+		if ( value === 'deleteOption' ) {
+			onDelete();
+		} else if ( value === 'settingsOption' ) {
+			openGeneralSidebar();
+		} else if ( value === 'forwardButtonOption' ) {
+			onMoveDown();
+		} else if ( value === 'backwardButtonOption' ) {
+			onMoveUp();
+		}
+	}
+
+	const disabledButtonIndices = options
+		.map( ( option, index ) => option.disabled && index + 1 )
+		.filter( Boolean );
+
+	const accessibilityHintIOS = __(
+		'Double tap to open Action Sheet with available options'
+	);
+	const accessibilityHintAndroid = __(
+		'Double tap to open Bottom Sheet with available options'
+	);
+
+	return (
+		<>
+			<ToolbarButton
+				title={ __( 'Open Block Actions Menu' ) }
+				onClick={ () => this.picker.presentPicker() }
```

Comment:
```suggestion
				onClick={ this.picker.presentPicker }
```

Reply:
Going to move it to the function with checking if `picker` ref exists.